### PR TITLE
9 Topics

### PR DESCRIPTION
**Pull request by HarCass**

### CODE REVIEW

File: src/api.js [hunk 1]
```diff
@@ -4,8 +4,8 @@ const newsApi = axios.create({
     baseURL: 'https://hc-nc-news-api.onrender.com/api'
 });
 
-export const getArticles = (page) => {
-    return newsApi.get('/articles', { params: {p: page} })
+export const getArticles = (page, topic) => {
+    return newsApi.get('/articles', { params: {p: page, topic} })
```
Comment (callumu21) [inline src/api.js:8]: Really nice use of Axios to control your optional query parameters here

File: src/hooks/useTopics.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { useEffect, useState } from "react"
```
Comment (callumu21) [inline src/hooks/useTopics.js:1]: Nice use of a custom React hook

**Review by callumu21 — CHANGES_REQUESTED** [review]
Really nice job, Harry. You've opted for an new endpoint rather than adding a topic query to your /articles route, but it looks as though the user interface for both displays is largely the same. This means your Articles and Topic components have a lot of the same repeated logic. Can you refactor to rely on a single component instead and then resubmit this PR for approval 👍

File: src/App.jsx [hunk 3]
```diff
@@ -13,6 +15,8 @@ function App() {
         <Route path="/" element={<Home></Home>}></Route>
         <Route path="/articles" element={<Articles></Articles>}></Route>
         <Route path="/articles/:article_id" element={<Article></Article>}></Route>
+        <Route path="/topics" element={<Topics></Topics>}></Route>
+        <Route path="/topics/:topic_name" element={<Topic></Topic>}></Route>
```
Comment (Pipe-Smoking-Rabbit) [inline src/App.jsx:19]: nicely done! It's always nice to be able to copy + paste links directly to a filtered page I find

**Review by Pipe-Smoking-Rabbit — APPROVED** [review]
Looks good to merge Harry! 👍

**Pull request merged**